README.md

```diff
@@ -231,172 +257,271 @@ curl -s -i https://api.travis-ci.org/auth/github \
+
+The following `script.sh` contains the script code for packaging your system's `libpng`. (Note: To understande this code, make sure to review the [Script structure](#script-structure) section.)
 
 ```bash
-cp -r scripts/libpng/1.6.28 scripts/yourlib/1.0.0
+#!/usr/bin/env bash
+
+MASON_NAME=libpng
+MASON_VERSION=system
+MASON_SYSTEM_PACKAGE=true
+
+. ${MASON_DIR}/mason.sh
+
+if [ ! $(pkg-config libpng --exists; echo $?) = 0 ]; then
+    mason_error "Cannot find libpng with pkg-config"
+    exit 1
+fi
+
+function mason_system_version {
+    mkdir -p "${MASON_PREFIX}"
+    cd "${MASON_PREFIX}"
+    if [ ! -f version ]; then
+        echo "#include <png.h>
+#include <stdio.h>
+#include <assert.h>
+int main() {
+    assert(PNG_LIBPNG_VER == png_access_version_number());
+    printf(\"%s\", PNG_LIBPNG_VER_STRING);
+    return 0;
+}
+" > version.c && ${CC:-cc} version.c $(mason_cflags) $(mason_ldflags) -o version
+    fi
+    ./version
+}
+
+function mason_compile {
+    :
+}
+
+function mason_cflags {
+    pkg-config libpng --cflags
+}
+
+function mason_ldflags {
+    pkg-config libpng --libs
+}
+
+mason_run "$@"
 ```
 
-See the [Script structure](#script-structure) section below for details on the `script.sh` format.
+System packages are marked with `MASON_SYSTEM_PACKAGE=true`. We're also first using `pkg-config` to check whether the library is present at all. The `mason_system_version` function creates a small executable which outputs the actual version. It is the only thing that is cached in the installation directory.
+
+We have to override the `mason_cflags` and `mason_ldflags` commands since the regular commands return flags for static libraries, but in the case of system packages, we want to dynamically link against the package.
+
+## Releasing a package
+
+Here is an example workflow to help get you started:
+
+1. Create an annotated tag in git for the code you want to package.
+
+    Annotated tags can be stored, checksummed, signed and verified with GNU Privacy Guard (GPG) in Github. To create an annotated tag specify `-a` when running the `tag` command, for example:
+
+    `git tag -a v0.1.0 -m "my version 0.1.0"`
+
+2. Share your new tag.
+
+    You have to explicitly push your new tag to a shared Github server. This is the location we share with Mason when specifying where to download the code to be packaged. Using our example above, we would run:
+
+    `git push origin v0.1.0`
+
+    Or you can push all tags:
+
+    `git push --tags`
+
+3. Create a package.
+
+    We recommend working in a new branch before creating a pacakge.
+
+    `git checkout -b my-version-0.1.0`
+
+    Follow the instructions in the [Getting started](#getting-started) section for creating a new package.
+
+4. Build your package.
```

Comment:
Worth noting that the `./mason build <package> <version>` will output a checksum error? I find myself searching for these docs pretty regularly.

Reply:
@mapsam - If the *Getting started* instructions are followed in step 3 then they should not run into that error. It's also mentioned there how we use that error to get the correct checksum. Looking forward to Mason v2 where I imagine we won't have to do this step. :)

Reply:
If step 3 is followed, in particular the **Getting started** instructions that step 3 links to, this is unnecessary.

Reply:
👍 didn't see it linked to the "getting started" section - sounds good!